service/controller/clusterapi/crds.go

```diff
@@ -0,0 +1,95 @@
+package clusterapi
+
+import (
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewClusterCRD returns a new custom resource definition for Cluster (from
+// Cluster API). This might look something like the following.
+//
+//     apiVersion: apiextensions.k8s.io/v1beta1
+//     kind: CustomResourceDefinition
+//     metadata:
+//       name: clusters.cluster.k8s.io
+//     spec:
+//       group: cluster.k8s.io
+//       scope: Namespaced
+//       version: v1alpha1
+//       names:
+//         kind: Cluster
+//         plural: clusters
+//         singular: cluster
+//       subresources:
+//         status: {}
+//
+func NewClusterCRD() *apiextensionsv1beta1.CustomResourceDefinition {
```

Comment:
Why don't we have this in `apiextensions`? I think it is wrong here. Imagine we introduce the Cluster API types in Azure and KVM, what then?

Reply:
I was thinking about that, but I felt it was wrong to have these in `apiextensions` when the types aren't defined there. Which package would you put them there? I'm fine moving these but I thought initially that it's not semantically right.

Reply:
I agree this should land in `apiextensions`. My suggestion is `cluster/v1alpha1`. Even though we don't have types there we still have provider extensions.